client/ipa-join.c

```diff
@@ -612,161 +632,417 @@ join_krb5(const char *ipaserver, char *hostname, char **hostdn, const char **pri
     return rval;
 }
 
+#else
+
+static inline struct curl_slist *
+curl_slist_append_log(struct curl_slist *list, char *string, bool quiet) {
+    list = curl_slist_append(list, string);
+    if (!list) {
+        if (!quiet)
+            fprintf(stderr, _("curl_slist_append() failed for value: '%s'\n"), string);
+        return NULL;
+    }
+    return list;
+}
+
+#define CURL_SETOPT(curl, opt, val) \
+    if (curl_easy_setopt(curl, opt, val) != CURLE_OK) { \
+        if (!quiet) \
+            fprintf(stderr, _("curl_easy_setopt() failed\n")); \
+        rval = 17; \
```

Comment:
What's the meaning of ``17`` here? How about you define global exist code as constants and reuse the named constants?

Reply:
The current join code is filled with numeric return values. 17 is XMLRPC fault in the man page, we could convert it to RPC fault. I'd be ok leaving the numeric values and fixing the constants later as it isn't related to the JSON code.

Reply:
I would prefer as well to do this separately. I think this PR is already big enough as it is.